asm/f64/asm_test.go

```diff
@@ -65,3 +68,48 @@ func nonStridedWrite(x []float64, inc int) bool {
 	}
 	return false
 }
+
+type incSet struct {
+	x, y int
+}
+
+func genInc(inc []int) []incSet {
+	n := len(inc)
+	is := make([]incSet, n*n)
+	for x := range inc {
+		for y := range inc {
+			is[x*n+y] = incSet{inc[x], inc[y]}
+		}
+	}
+	return is
+}
+
+type incToSet struct {
+	dst, x, y int
+}
+
+func genIncTo(inc []int) []incToSet {
+	n := len(inc)
+	is := make([]incToSet, n*n)
+	for dst := range inc {
+		for x := range inc {
+			for y := range inc {
+				is[x*n+y] = incToSet{inc[dst], inc[x], inc[y]}
+			}
+		}
+	}
+	return is
+}
+
+var gs []float64
+
+func randomSlice(n, inc int) []float64 {
```

Comment:
This is only used in Dscal* benchmarks and will be disappearing when I re-work that code, because it's not serving any purpose there that isn't better served by the global benchmark slices. 

I only moved it because I was removing it from the Axpy* benchmarks, for the same reason.